app/controllers/key_pairs_controller.rb

```diff
@@ -38,36 +51,99 @@ def edit
 
   # Given a File object to upload, save it on the file system with
   # association to the user_name given.
-  # Admins are able to save keys under different user names then
-  # their own.
+  # If a String is supplied as the first argument then it's content
+  # is used to create the public key
   # Creates the KEY_STORAGE directory if it does not yet exist
-  def upload_key_file(_file, user_name)
+  def upload_key_file(_file, user_name, time_stamp)
     Dir.mkdir(KEY_STORAGE) unless File.exists?(KEY_STORAGE)
-    uploaded_io = _file
 
-    # todo: check current user to see if they have permission to
-    # use a different user_name then their current one (admins)
+    public_key_content = ''
+
+    if _file.is_a? String
+      # Dump the string into the file
+      public_key_content = _file
+    else
+      public_key_content = _file.read
+    end
 
-    File.open(Rails.root.join(KEY_STORAGE, user_name + '.pub'), 'wb') do |f|
-      f.write(uploaded_io.read)
+    File.open(Rails.root.join(KEY_STORAGE, user_name + "@" + time_stamp + '.pub'), 'wb') do |f|
```

Comment:
Line is too long. [95/80]<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.